utils/doclint/xmlDocumentation.js

```diff
@@ -0,0 +1,197 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// @ts-check
+
+const Documentation = require('./documentation');
+const { visitAll } = require('../markdown');
+/**
+ * @param {Documentation.MarkdownNode[]} nodes
+ * @param {number} maxColumns
+ */
+function renderXmlDoc(nodes, maxColumns) {
+    const summary = [];
+    const examples = [];
+    /** @type {Documentation.MarkdownNode} */
+    let lastNode;
+
+    summary.push('<summary>');
+
+    visitAll(nodes, (node, depth) => {
+        lastNode = innerRenderXmlNode(node, lastNode, summary, examples, maxColumns);
+    });
+
+    if (summary.length == 1) { // just the <summary> node 
+        return [];
+    }
+
+    // we might have a stray list, if the <li> is the last element
+    if (lastNode && lastNode.type === 'li') {
+        summary.push('</list>');
+    }
+    summary.push('</summary>');
+
+    // add examples
+    summary.push(...examples);
+    return summary.map(n => `/// ${n}`);
+}
+
+/**
+ * @param {Documentation.MarkdownNode} node
+ * @param {Documentation.MarkdownNode} lastNode
+ * @param {number=} maxColumns
+ * @param {string[]} summary
+ * @param {string[]} examples
+ */
+function innerRenderXmlNode(node, lastNode, summary, examples, maxColumns) {
+    /** @param {string[]} a */
+    const newLine = (a) => {
+        if (a[a.length - 1] !== '')
+            a.push('');
+    };
+
+    let escapedText = node.text;
+    // resolve links (within [])
+
+    if (node.type === 'text') {
+        // clear up the list, if there was one
+        if (lastNode && lastNode.type === 'li') {
+            summary.push('</list>');
+        }
+
+        summary.push(...wrapText(escapedText, maxColumns));
+
+        return node;
+    }
+
+    if (node.type === 'li') {
+        if (escapedText.startsWith('extends: ')) {
+            summary.push(...wrapText(`<seealso cref="${escapedText.substring(9)}"/>`, maxColumns));
+            return undefined;
+        }
+
+        // if the previous node was not li, start list
+        if (lastNode && lastNode.type !== 'li') {
+            summary.push(`<list>`);
+        }
+
+        summary.push(...wrapText(`<item><description>${escapedText}</description></item>`, maxColumns));
+    }
+
+    // TODO: this should really move to the examples array
+    if (node.type == 'code' && node.codeLang == "csharp") {
+        summary.push('<![CDATA[');
+        summary.push(...node.lines);
+        summary.push(']]>');
+    }
+
+    return node;
+}
+
+/**
+ * @param {string} text
+ * @param {number=} maxColumns
+ * @param {string=} prefix
+ */
+function wrapText(text, maxColumns = 0, prefix = '') {
```

Comment:
It looks like you are inserting the xml tags and then fighting with it. I'd suggest that you only wrap user content instead. I.e. instead of
```
wrapText(`<item><description>${escapedText}</description></item>`, maxColumns)
```

do
```
`<item><description>\n${wrapText(escapedText, maxColumns)}\n</description></item>`)
```

That way your wrapText becomes trivial.